esp/src/src/ESPPreflight.ts

```diff
@@ -1,10 +1,260 @@
+            case "ThorCluster":
+                return "THORMACHINES"
+        }
+    },
+
+    getOS: function (int) {
+        switch (int) {
+            case 0:
+                return "Windows"
+            case 1:
+                return "Solaris"
+            case 2:
+                return "Linux"
+            default:
+                return "Linux"
+        }
+    }
+})
+
+var ClusterProcessStore = declare([ESPRequest.Store], {
+    service: "WsTopology",
+    action: "TpClusterQuery",
+    responseQualifier: "TpClusterQueryResponse.TpClusters.TpCluster",
+    idProperty: "hpcc_id",
+    constructor: function (options) {
+        if (options) {
+            declare.safeMixin(this, options);
+        }
+    },
+    preProcessRow: function (row) {
+        lang.mixin(row, {
+            Platform: this.getOS(row.OS),
+            hpcc_id: row.Name,
+            displayName: row.Name,
+            type: "clusterProcess",
+            Component: row.Type,
+            Configuration: true
+        });
+    },
+    mayHaveChildren: function (item) {
+        return item.type === "clusterProcess";
+    },
+    getChildren: function (parent, options) {
+        var store = Observable(new ClusterProcessesList({
+            parent: parent
+        }));
+        return store.query({
+            Type: this.getMachineType(parent.Type),
+            Parent: parent,
+            Cluster: parent.Name,
+            LogDirectory: parent.LogDirectory,
+            Path: parent.Path,
+            Directory: parent.Directory
+        });
+    },
+
+    getMachineType: function (type) {
+        switch (type) {
+            case "RoxieCluster":
+                return "ROXIEMACHINES"
+            case "ThorCluster":
+                return "THORMACHINES"
+        }
+    },
+
+    getOS: function (int) {
+        switch (int) {
+            case 0:
+                return "Windows"
+            case 1:
+                return "Solaris"
+            case 2:
+                return "Linux"
+            default:
+                return "Linux"
+        }
+    }
+});
+
+var ClusterProcessesList = declare([ESPRequest.Store], {
+    service: "WsTopology",
+    action: "TpMachineQuery",
+    responseQualifier: "TpMachineQueryResponse.TpMachines.TpMachine",
+    idProperty: "hpcc_id",
+
+    preProcessRow: function (row) {
+        lang.mixin(row, {
+            Platform: this.getOS(row.OS),
+            hpcc_id: row.Name + "_" + row.ProcessNumber,
+            displayName: row.Name,
+            type: "machine",
+            Component: row.Type,
+            Domain: row.Domain,
+            Directory: "",
+            Parent: this.parent
+        });
+    },
 
+    getOS: function (int) {
+        switch (int) {
+            case 0:
+                return "Windows"
+            case 1:
+                return "Solaris"
+            case 2:
+                return "Linux"
+            default:
+                return "Linux"
```

Comment:
Minor:  Shorten to:
```
            case 2:
            default:
                return "Linux"
```

Reply:
Minor:  But if your making changes for the inherited init call, you may as well do this as well.